gandi/resource_glue_record.go

```diff
@@ -0,0 +1,150 @@
+package gandi
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/go-gandi/go-gandi/domain"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceGlueRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGlueRecordCreate,
+		Read:   resourceGlueRecordRead,
+		Update: resourceGlueRecordUpdate,
+		Delete: resourceGlueRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Domain name",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Host name of the glue record.",
+			},
+			"values": {
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "List of IP addresses.",
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{Default: schema.DefaultTimeout(1 * time.Minute)},
+	}
+}
+
+func resourceGlueRecordCreate(d *schema.ResourceData, meta interface{}) (err error) {
+	client := meta.(*clients).Domain
+	resDomain := d.Get("zone").(string)
+	name := d.Get("name").(string)
+
+	var values []string
+	for _, i := range d.Get("values").([]interface{}) {
+		values = append(values, i.(string))
+	}
+	sort.Strings(values)
+
+	request := domain.GlueRecordCreateRequest{
+		Name: name,
+		IPs:  values,
+	}
+
+	err = client.CreateGlueRecord(resDomain, request)
+	if err != nil {
+		return
+	}
+
+	// Sent, got 202 response. Delay read.
+	time.Sleep(2 * time.Second)
+
+	d.SetId(name)
+
+	return resourceGlueRecordRead(d, meta)
+}
+
+func resourceGlueRecordRead(d *schema.ResourceData, meta interface{}) (err error) {
+	client := meta.(*clients).Domain
+	resDomain := d.Get("zone").(string)
+
+	id :=d.Id()
+	records, err := client.ListGlueRecords(resDomain)
```

Comment:
You should add a `GetGlueRecords` method in the go-gandi lib instead. 

This would remove the 10lines below.

Reply:
Agreed.  Added here: https://github.com/go-gandi/go-gandi/pull/47

I'll update this PR when that gets merged to simplify this code and add the "data" object for the glue_record.

Reply:
that is merged ;)